callable/src/main/java/io/grpc/callable/Callable.java

```diff
@@ -0,0 +1,423 @@
+   * input. The supplied executor is used for creating tasks for each input. Example:
+   *
+   * <pre>
+   *  Callable.Transformer&lt;RequestT, ResponseT> transformer = ...;
+   *  Response response = Callable.create(transformer, executor).call(channel, request);
+   * </pre>
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(Transformer<? super RequestT, ? extends ResponseT> transformer, Executor executor) {
+    return new TransformingCallable<RequestT, ResponseT>(transformer, executor);
+  }
+
+
+  /**
+   * Returns a callable which executes the given function immediately on each provided input.
+   * Similar as {@link #create(Transformer, Executor)} but does not operate asynchronously and does
+   * not require an executor.
+   *
+   * <p>Note that the callable returned by this method does not respect flow control. Some
+   * operations applied to it may deadlock because of this. However, it is safe to use this
+   * callable in the context of a {@link #followedBy(Callable)} operation, which is the major
+   * use cases for transformers. But if you use a transformer to simulate a real rpc
+   * you should use {@link #create(Transformer, Executor)} instead.
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(Transformer<? super RequestT, ? extends ResponseT> transformer) {
+    return new TransformingCallable<RequestT, ResponseT>(transformer, null);
+  }
+
+  /**
+   * Interface for a transformer. It can throw a {@link StatusException} to indicate an error.
+   */
+  public interface Transformer<RequestT, ResponseT> {
+    ResponseT apply(RequestT request) throws StatusException;
+  }
+
+  /**
+   * Returns a callable which echos its input.
+   */
+  public static <RequestT> Callable<RequestT, RequestT>
+      identity() {
+    return new TransformingCallable<RequestT, RequestT>(new Transformer<RequestT, RequestT>() {
+      @Override public RequestT apply(RequestT request) throws StatusException {
+        return request;
+      }
+    }, null);
+  }
+
+  /**
+   * Returns a callable which always returns the given constant.
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      constant(final ResponseT result) {
+    return new TransformingCallable<RequestT, ResponseT>(new Transformer<RequestT, ResponseT>() {
+      @Override public ResponseT apply(RequestT request) throws StatusException {
+        return result;
+      }
+    }, null);
+  }
+
+  // Followed-By
+  // ===========
+
+  /**
+   * Returns a callable which forwards the responses from this callable as requests into the other
+   * callable. Works both for unary and streaming operands. Example:
+   *
+   * <pre>
+   * String bookName = ...;
+   * Callable.Transformer&lt;Book, GetAuthorRequest> bookToGetAuthorRequest = ...;
+   * Author response =
+   *     Callable.create(LibraryGrpc.CONFIG.getBook)
```

Comment:
My guideline here is it something we can imagine to execute on server side. Followedby and transform (with a restricted transform function) seems to me. I agree we should not re-invent all the combinators of Rx/Java 8 here.